apache/superset
# refactor: add "button" role to clickable UI elements for improved accessibility

### SUMMARY

Clickable UI elements lacking `<a>` or `<button>` tags, or the `role="button"` attribute, weren't semantically clear as clickable elements when using a screen reader. 

To improve accessibility, I reviewed the React codebase for `onClick` handlers. When encountering HTML elements with `onClick` handlers AND without `<a>` or `<button>` tags, I added the `role="button"` attribute to identify them as clickable. 

This approach targets the most immediate and straightforward cases, ensuring that elements triggered by onClick handlers are now clearly identified as clickable for users relying on screen readers. 

### TESTING INSTRUCTIONS

Testing options include:
- Using a screen reader: identify and navigate through the clickable components and ensure they are announced correctly 
- Using browser devtools: inspect clickable elements and verify that elements without `<a>` or `<button>` tags have the `role="button"` attribute.



### ADDITIONAL INFORMATION


- [x] Has associated issue: #26190
